janus_server/src/datastore.rs

```diff
@@ -1163,6 +1163,44 @@ impl Transaction<'_> {
         }))
     }
 
+    /// Returns the number of rows in `aggregate_share_jobs` whose batch interval contains
+    /// `batch_unit_interval`.
+    #[tracing::instrument(skip(self), err)]
+    pub(crate) async fn get_aggregate_share_job_count_by_batch_unit(
+        &self,
+        task_id: TaskId,
+        batch_unit_interval: Interval,
+    ) -> Result<u64, Error> {
+        let unit_interval_start = batch_unit_interval.start().as_naive_date_time();
+        let unit_interval_end = batch_unit_interval.end().as_naive_date_time();
+
+        let stmt = self
+            .tx
+            .prepare_cached(
+                "SELECT COUNT(*) FROM aggregate_share_jobs
+                WHERE
+                    task_id = (SELECT id FROM tasks WHERE task_id = $1)
+                    AND batch_interval_start <= $2
+                    AND (batch_interval_start + batch_interval_duration * interval '1 second') >= $3",
+            )
+            .await?;
+        let row = self
+            .tx
+            .query_one(
+                &stmt,
+                &[
+                    /* task_id */ &task_id.as_bytes(),
+                    &unit_interval_start,
+                    &unit_interval_end,
+                ],
+            )
+            .await?;
+
+        // PostgreSQL's `COUNT` aggregate function returns rows with a bigint-valued "count" column.
```

Comment:
You can also name the returned column (as in `SELECT COUNT(*) AS count FROM...`) to remove the ambiguity.